Dynamic protocols switching (#113)
nrfconnect/ncs-zigbee: docs/includes/matter_extension_protocol_switch.txt

The default protocol on a factory-fresh device is selected by the ``CONFIG_ZIGBEE_MATTER_PROTOCOL_STATE_DEFAULT_PROTOCOL`` Kconfig choice (Zigbee by default).
After the first run, the persisted protocol state in settings takes precedence.

When ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_BUTTON_SWITCH`` is enabled (default), a long press on the selected button switches the active protocol.
The required hold time is configured by ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_SWITCH_BUTTON_PRESS_TIME_SECONDS`` (5 s by default).
From Zigbee, the coexistence layer stops the ZBOSS stack and hands the 802.15.4 radio to OpenThread for Matter.
From Matter, the device persists Zigbee as the active protocol and reboots.

> The switch is ignored while Zigbee network join commissioning or Matter pairing is in progress.

Reviewer: ```suggestion
The switch is ignored while Zigbee network joining or Matter commissioning is in progress.
```
Maybe in this way? But I'm not sure.

The default protocol on a factory-fresh device is selected by the ``CONFIG_ZIGBEE_MATTER_PROTOCOL_STATE_DEFAULT_PROTOCOL`` Kconfig choice (Zigbee by default).
A Matter factory reset (including last-fabric removal when ``CONFIG_CHIP_LAST_FABRIC_REMOVED_ERASE_AND_REBOOT`` is enabled) applies the same default again and wipes Matter commissioning data.
After the first run, the persisted protocol state in settings takes precedence until a factory reset or explicit protocol switch changes it.

> When ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_BUTTON_SWITCH`` is enabled (default), a long press on the selected button switches the active protocol.

Reviewer 2: ```suggestion
When ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_BUTTON_SWITCH`` is enabled (the default), you can long-press the selected button to switch the active protocol.
```

> The required hold time is configured by ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_SWITCH_BUTTON_PRESS_TIME_SECONDS`` (5 s by default).

Reviewer 2: ```suggestion
Configure the required hold time by setting the ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_SWITCH_BUTTON_PRESS_TIME_SECONDS`` Kconfig option. 
By default, this option is set to 5 seconds.
```

From Zigbee, the coexistence layer stops the ZBOSS stack and hands the 802.15.4 radio to OpenThread for Matter.
From Matter, the device persists Zigbee as the active protocol and reboots.

> The switch is ignored while Zigbee network joining or Matter commissioning is in progress.

Reviewer 2: ```suggestion
The protocol switch is ignored while the device is joining a Zigbee network or undergoing Matter commissioning.
```

Other Kconfig options in the coexistence layer:

> * ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_CHIP_BLE_WHILE_ZIGBEE`` (default enabled) — Matter may advertise for commissioning over Bluetooth LE while Zigbee owns the 802.15.4 radio.

Reviewer 3: Minor: this `CHIP` is leftover after very old name of Matter protocol (Connected Home over IP) and we prefer to not use it anymore. Also BLE should not be used as shortcut. How about calling it like  `CONFIG_ZIGBEE_MATTER_COEXISTENCE_BT_ADV_WHILE_ZIGBEE`?

  When disabled, the sample must call ``zigbee_matter_coexistence_signal_matter_board_init()`` after Matter server init, and the Zigbee worker proceeds without waiting for CHIPoBLE advertising.

> * ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_BUTTON_SWITCH`` — enable or disable the button-triggered protocol switch described above.

Reviewer 2: To avoid "before, above, below"
```suggestion
* ``CONFIG_ZIGBEE_MATTER_COEXISTENCE_BUTTON_SWITCH`` — Enable or disable the button-triggered protocol switch.
```
